governance/program/src/state/proposal.rs

```diff
@@ -229,25 +280,84 @@ impl Proposal {
         Ok(())
     }
 
+    /// Returns final proposal state after vote ends
     fn get_final_vote_state(
         &mut self,
         max_vote_weight: u64,
         config: &GovernanceConfig,
-    ) -> ProposalState {
-        let yes_vote_threshold_count =
-            get_yes_vote_threshold_count(&config.vote_threshold_percentage, max_vote_weight)
+    ) -> Result<ProposalState, ProgramError> {
+        // Get the min vote weight required for options to pass
+        let min_vote_threshold_weight =
+            get_min_vote_threshold_weight(&config.vote_threshold_percentage, max_vote_weight)
                 .unwrap();
 
-        // Yes vote must be equal or above the required yes_vote_threshold_percentage and higher than No vote
-        // The same number of Yes and No votes is a tie and resolved as Defeated
-        // In other words  +1 vote as a tie breaker is required to Succeed
-        if self.yes_votes_count >= yes_vote_threshold_count
-            && self.yes_votes_count > self.no_votes_count
-        {
-            ProposalState::Succeeded
-        } else {
+        // If the proposal has a reject option then any other option must beat it regardless of the configured min_vote_threshold_weight
+        let deny_vote_weight = self.deny_vote_weight.unwrap_or(0);
+
+        let mut best_succeeded_option_weight = 0;
+        let mut best_succeeded_option_count = 0;
```

Comment:
No, I've kept it as it was but added more comments to make it clearer.
There can be several successful options (`best_succeeded_option_count`) with the highest score (`best_succeeded_option_weight`) and when we have a single choice vote then the proposal is `Defeated` because there can only be single  winning option